CLI tools/MultiMC to Git.py

```diff
@@ -7,24 +7,20 @@
 
 # Functions
 
-def copy_file(from_path, to_path, from_desc, to_desc):
+def copy_file(from_path: str, to_path: str, from_desc: str, to_desc: str) -> None:
     if os.path.isfile(from_path):
         shutil.copy2(from_path, to_path)
         print("Copied " + from_desc + " to " + to_desc)
     else:
         print("Skipped " + from_desc + " copying to " + to_desc + ", didn't exist")
 
-def copy_to_archive(from_path, to_path, archive_path):
-
-    files = []
+def copy_to_archive(from_path: str, to_path: str, archive_path: str) -> None:
 
     with zipfile.ZipFile(archive_path) as archive:
-        for zipinfo in archive.infolist():
-            if zipinfo.filename != to_path:
-                files.append((zipinfo.filename, archive.read(zipinfo.filename)))
+        files = {zipinfo.filename: archive.read(zipinfo.filename) for zipinfo in archive.infolist() if zipinfo.filename != to_path}
 
     with zipfile.ZipFile(archive_path, "w") as archive:
-        for filename, content in files:
+        for filename, content in files.items():

```

Comment:
Python doesn't optimize it automatically?

Reply:
It's not optimizing anything, just making things more inline and messy. When I wrote this, I was aiming for readability and fitting existing code style. But this... I dunno, looks nasty.

Reply:
> I thought it chose the type by the contents.

He changed the type of var `files` from list to dictionary, and you can't just iterate over dict. You can iterate over `.values()` of this dict, or over `'.items()` which is list KeyValue's